### PR TITLE
Support multiple rdg views

### PR DESCRIPTION
**Pull request by e-mcginnis**
### PR Checklist

* [*] test-datasets is a publicly accessible github repo. The contents of this PR can be made public.
* [*] The naming and organization of the dataset I am adding follows https://github.com/KatanaGraph/test-datasets#organization
* [*] If I am adding an RDG, its directory contains a (import/migrate/generate) script, and a README describing the RDGs usage. https://github.com/KatanaGraph/test-datasets#the-following-scripts-are-conditionally-present-for-each-rdg-one-of-the-following-options-must-be-present
* [*] If I am uprevving the storage_format_version of the rdgs in this repo, I have followed https://github.com/KatanaGraph/test-datasets#how-to-uprev-the-rdgs-in-this-repo

### List of folks that should be notified of this PR. Please don't remove these. 
@scober
@aneeshdurg

I suggest reviewing by commit, rather than by the entire PR. Only one of the commits actually contains code changes: "Support rdgs with multiple format views"

```
new RDG storage format versioning is of the form:
  DLSG#.STPG#

0 means a view is not available on that rdg
```

### CODE REVIEW

File: rdg_datasets/rmat15_hosts=2_policy=roec/generate.py [hunk 1]
```diff
@@ -12,17 +12,24 @@
 num_partitions = 2
 generate_args = ["--partition=retained-oec"]
 
+def uprev(config: Config, new_rdg_storage_format_version: int) -> pathlib.Path:
+    #TODO(emcginnis) we can't generate a partitioned rdg since lspg only fully supports random-oec
+    # use migrate instead for now
+    # once we can partition as the correct scheme, uncomment this and remove the migrate_dist call
```
[the remark this answers is not in the document]
Comment (e-mcginnis) [reply rdg_datasets/rmat15_hosts=2_policy=roec/generate.py:18]: sure, I can add something like that

**Comment by tylershunt** [comment]
That's a lot of files!
image

**Comment by e-mcginnis** [comment]
> That's a lot of files! image

I have truly peaked :infinity:

**Comment by e-mcginnis** [comment]
once https://github.com/KatanaGraph/test-datasets/pull/30 merges, this PR will have to be fixed up to include that new RDG at the new storage format version using:
```
uprev rdgs --rdg_storage_format_version="DLSG1.STPG9" --build_dir="~/katana/katana-enterprise/build" --rdg=smiles_targets_predictions_small
```

**Comment by tylershunt** [comment]
Hey @scober do we still need this?

**Comment by scober** [comment]
@tylershunt my personal preference would be to have a single view per directory. We do need some of the things covered in this PR but my guess is that this PR is too stale to be useful.

**Comment by tylershunt** [comment]
Closing because of staleness.

**Pull request closed without merging**